description: owners.py is not in the -lite tarball, remove function that uses it
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/flags/list_flags.py
+++ b/tools/flags/list_flags.py
@@ -21,7 +21,6 @@ sys.path.append(PYJSON5_PATH)
 sys.path.append(DEPOT_TOOLS_PATH)
 
 import json5
-import owners
 
 
 def load_metadata():
@@ -55,35 +54,6 @@ def keep_never_expires(flags):
   return [f for f in flags if f['expiry_milestone'] == -1]
 
 
-def resolve_owners(flags):
-  """Resolves sets of owners for every flag in the provided list.
-
-  Given a list of flags, for each flag, resolves owners for that flag. Resolving
-  owners means, for each entry in a flag's owners list:
-  * Turning owners files references into the transitive set of owners listed in
-    those files
-  * Turning bare usernames into @chromium.org email addresses
-  * Passing any other type of entry through unmodified
-  """
-
-  owners_db = owners.Database(ROOT_PATH, open, os.path)
-
-  new_flags = []
-  for f in flags:
-    new_flag = f.copy()
-    new_owners = []
-    for o in f['owners']:
-      if o.startswith('//') or '/' in o:
-        new_owners += owners_db.owners_rooted_at_file(re.sub('//', '', o))
-      elif '@' not in o:
-        new_owners.append(o + '@chromium.org')
-      else:
-        new_owners.append(o)
-    new_flag['resolved_owners'] = new_owners
-    new_flags.append(new_flag)
-  return new_flags
-
-
 def print_flags(flags, verbose):
   """Prints the supplied list of flags.
 
--- a/tools/metrics/histograms/histograms.xml
+++ b/tools/metrics/histograms/histograms.xml
@@ -4784,7 +4784,6 @@ reviews. Googlers can read more about th
 <histogram name="Android.WebView.AndroidX.ApiCall" enum="AndroidXWebkitApiCall"
     expires_after="2021-01-29">
   <owner>laisminchillo@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records calls to WebView APIs from AndroidX. Some AndroidX-originating calls
     will plumb through the framework, for example, if methods that are available
@@ -4876,7 +4875,6 @@ reviews. Googlers can read more about th
     expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the number of times the developer UI app was launched. This only
     ever logs the value &quot;true&quot; because we only care about the total
@@ -4888,7 +4886,6 @@ reviews. Googlers can read more about th
     enum="AndroidWebViewCrashCollectionState" expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the state of crash collection, as seen by the developer UI. This is
     recorded when launching the crash UI, but only if the UI is not already
@@ -4903,7 +4900,6 @@ reviews. Googlers can read more about th
     units="crash reports" expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the number of crashes shown in the developer UI. This is logged each
     time the crash UI is launched and each time its list is refreshed. The UI
@@ -4916,7 +4912,6 @@ reviews. Googlers can read more about th
     expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the time spent blocking WebView startup to check for developer mode
     (see &quot;Android.WebView.DevUi.DeveloperModeEnabled&quot;). This is
@@ -4929,7 +4924,6 @@ reviews. Googlers can read more about th
     enum="BooleanEnabled" expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records whether or not WebView is starting up in &quot;developer mode.&quot;
     This is a mode where the user has enabled debugging features through the
@@ -4941,7 +4935,6 @@ reviews. Googlers can read more about th
     expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the time spent blocking WebView startup to load flags from the
     developer UI. This is recorded only once during WebView startup, and only if
@@ -4957,7 +4950,6 @@ reviews. Googlers can read more about th
 
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the user's navigation to a Fragment. &quot;FromIntent&quot; tracks
     navigations triggered by an Intent (ex. navigating to HomeFragment on
@@ -4970,7 +4962,6 @@ reviews. Googlers can read more about th
     enum="AndroidWebViewDevUiMenu" expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the user's selection in the developer UI overflow menu or action
     bar, recorded after each selection.
@@ -4981,7 +4972,6 @@ reviews. Googlers can read more about th
     enum="BooleanEnabled" expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records how the developer UI launcher icon for Monochrome was toggled: was
     it enabled or disabled? This can be recorded either from WebView's
@@ -4999,7 +4989,6 @@ reviews. Googlers can read more about th
 
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the time spent using a specific tool, from creation to destruction.
   </summary>
@@ -5009,7 +4998,6 @@ reviews. Googlers can read more about th
     expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the number of flags we will toggle for this run of WebView. This is
     recorded only once during WebView startup, and only if we've already
@@ -5057,7 +5045,6 @@ reviews. Googlers can read more about th
     enum="WebViewUrlScheme" expires_after="2021-05-05">
   <owner>ntfschr@chromium.org</owner>
   <owner>torne@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the scheme for the baseUrl parameter to loadDataWithBaseURL. This
     also records if this value is &quot;empty&quot;, as determined by
@@ -5092,7 +5079,6 @@ reviews. Googlers can read more about th
     expires_after="2021-05-05">
   <owner>ntfschr@chromium.org</owner>
   <owner>torne@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the scheme for the URL loaded by a browser-initiated navigation.
     Specifically, this is the scheme of the URL passed to loadUrl() (and its
@@ -5109,7 +5095,6 @@ reviews. Googlers can read more about th
     keep around.
   </obsolete>
   <owner>ntfschr@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records whether WebView needed to backfill the install date pref with
     Android frameworks APIs. Recorded once during startup (metrics service
@@ -5140,7 +5125,6 @@ reviews. Googlers can read more about th
     units="histograms" expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the number of histograms received from WebView's non-embedded
     histogram service. This is recorded once during WebView startup, after
@@ -5152,7 +5136,6 @@ reviews. Googlers can read more about th
     enum="AndroidWebViewDevUiMetricsTransmission" expires_after="2021-01-20">
   <owner>ntfschr@chromium.org</owner>
   <owner>hazems@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records whether metrics transmission from the non-embedded histogram service
     was successful. This is recorded once during WebView startup, after
@@ -5211,7 +5194,6 @@ reviews. Googlers can read more about th
 <histogram name="Android.WebView.PrefLoadTime" units="ms"
     expires_after="2020-12-01">
   <owner>rmcelrath@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records how long WebView blocks startup to load prefs from disk. This is
     recorded once per browser process startup.
@@ -5328,7 +5310,6 @@ reviews. Googlers can read more about th
 <histogram name="Android.WebView.Startup.DataDirLockAttempts" units="attempts"
     expires_after="2020-09-30">
   <owner>torne@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     How many attempts it took to acquire the WebView data directory lock. A
     value of 0 indicates that we reached the max retry count and failed to
@@ -5339,7 +5320,6 @@ reviews. Googlers can read more about th
 <histogram name="Android.WebView.Startup.InitAlreadyStarted" enum="Boolean"
     expires_after="2020-09-30">
   <owner>torne@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records whether we already tried to initialize WebView in the current
     process previously, upon entry to the initialization function. This should
@@ -5352,7 +5332,6 @@ reviews. Googlers can read more about th
     enum="WebViewClientTypeEnum" expires_after="2021-05-05">
   <owner>laisminchillo@chromium.org</owner>
   <owner>ntfschr@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records whether the WebViewClient provided by the app is an instance of
     WebViewClientCompat (and therefore exposes AndroidX callbacks). This is
@@ -5388,7 +5367,6 @@ reviews. Googlers can read more about th
 <histogram name="Android.WebView.Visibility.Global" enum="WebViewVisibility"
     expires_after="2021-01-01">
   <owner>idries@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records the global visibility state of WebView in seconds. The Visible
     bucket counts total seconds that any WebView was visible (the view itself
@@ -5402,7 +5380,6 @@ reviews. Googlers can read more about th
 <histogram name="Android.WebView.Visibility.PerWebView"
     enum="WebViewVisibility" expires_after="2021-01-01">
   <owner>idries@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Records per-WebView visibility duration in seconds. The Visible bucket
     counts the total time in seconds that each WebView was considered visible
@@ -29845,7 +29822,6 @@ reviews. Googlers can read more about th
 <histogram name="ContextMenu.CancelSystemTouches" enum="BooleanHit"
     expires_after="2020-12-11">
   <owner>michaeldo@chromium.org</owner>
-  <owner>src/ios/web/web_state/ui/OWNERS</owner>
   <summary>
     Logs true when the system touches are cancelled in order to show the Context
     Menu instead of performing a default system action. Only logged on iOS.
@@ -29855,7 +29831,6 @@ reviews. Googlers can read more about th
 <histogram name="ContextMenu.DelayedElementDetails"
     enum="ContextMenuDelayedElementDetails" expires_after="2020-12-11">
   <owner>michaeldo@chromium.org</owner>
-  <owner>src/ios/web/web_state/ui/OWNERS</owner>
   <summary>
     Logged when DOM element details are not yet available when the user
     completes a long press. Also logs if the context menu was shown once the
@@ -29883,7 +29858,6 @@ reviews. Googlers can read more about th
 <histogram name="ContextMenu.DOMElementFrame" enum="ContextMenuDOMElementFrame"
     expires_after="2020-12-11">
   <owner>michaeldo@chromium.org</owner>
-  <owner>src/ios/web/web_state/ui/OWNERS</owner>
   <summary>
     Logged when the Context Menu will be displayed. Also logs if the element for
     the last touch was found in the main frame or a iframe. Only logged on iOS.
@@ -30082,7 +30056,6 @@ reviews. Googlers can read more about th
 <histogram name="ContextMenu.WaitingForElementDetails" enum="BooleanHit"
     expires_after="2020-12-11">
   <owner>michaeldo@chromium.org</owner>
-  <owner>src/ios/web/web_state/ui/OWNERS</owner>
   <summary>
     Logs true when the context menu gesture recognizer fully recognizes a long
     press, but is still waiting on the DOM element details before the context
@@ -178768,7 +178741,6 @@ regressions. -->
     expires_after="2020-11-01">
   <owner>rmcelrath@chromium.org</owner>
   <owner>ntfschr@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     The time interval between when a WebView-using app's copy of the variations
     seed was written to its local app storage and when it was loaded for use.
@@ -178784,7 +178756,6 @@ regressions. -->
     expires_after="2020-11-01">
   <owner>rmcelrath@chromium.org</owner>
   <owner>ntfschr@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     The result of checking if a WebView-using app's copy of the variations seed
     is still fresh, or if a new seed should be requested.
@@ -178822,7 +178793,6 @@ regressions. -->
     has a higher maximum bucket.
   </obsolete>
   <owner>rmcelrath@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     The duration of the network request to downloads a new WebView variations
     seed from the variations server to the central on-device service.
@@ -179053,7 +179023,6 @@ regressions. -->
     expires_after="2020-11-01">
   <owner>rmcelrath@chromium.org</owner>
   <owner>ntfschr@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     Number of consecutive times WebView started up with a stale seed. Only
     written by WebView, when its loaded seed is fresh after previously being
@@ -179591,7 +179560,6 @@ regressions. -->
 <histogram name="Variations.WebViewDownloadJobFetchResult"
     enum="VariationsSeedFetchResult" expires_after="2021-01-31">
   <owner>rmcelrath@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     The result of a WebView seed download. Records both the HTTP code and
     various error values in one enumeration.
@@ -179601,7 +179569,6 @@ regressions. -->
 <histogram name="Variations.WebViewDownloadJobFetchTime2" units="ms"
     expires_after="2021-01-31">
   <owner>rmcelrath@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     The duration of the network request to downloads a new WebView variations
     seed from the variations server to the central on-device service.
@@ -179611,7 +179578,6 @@ regressions. -->
 <histogram name="Variations.WebViewDownloadJobInterval" units="minutes"
     expires_after="2021-01-31">
   <owner>rmcelrath@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     The time between the start of two consecutive WebView seed downloads by
     WebView's service. This metric will not be written if seeds are never
@@ -179622,7 +179588,6 @@ regressions. -->
 <histogram name="Variations.WebViewDownloadJobQueueTime" units="minutes"
     expires_after="2021-01-31">
   <owner>rmcelrath@chromium.org</owner>
-  <owner>src/android_webview/OWNERS</owner>
   <summary>
     The delay between when a WebView seed download was scheduled by the service
     with the JobScheduler and when it was executed.
